sample/SipInterconnect/README.md

```diff
@@ -0,0 +1,238 @@
+  - SIP headers to be added to the SIP INVITE request
+
+In this sample application, a SIP token is defined as part of the metadata, which may be be useful for embedding UI metadata for the audio participant. The token is generated as part of the token metadata (`data: "sip=true"`, which is stored in `event.stream.connection.data`) as the OpenTok session is created:
+
+```javascript
+opentok.createSession({ mediaMode:"routed" }, function(error, session) {
+  if (error) {
+    console.log("Error creating session:", error);
+    res.status(500).send("Error creating session.");
+  } else {
+    sessionId = session.sessionId;
+
+    //  Use the moderator role so you can force disconnect the SIP call
+    // Generate a token.
+    webrtcToken = opentok.generateToken(sessionId, {role: "moderator"});
+    sipToken = opentok.generateToken(sessionId, {data: "sip=true"});
+  }
+});
+```
+
+**NOTE:** Such metadata is not actually necessary to start a SIP call, though it has advantages as shown in this sample app. It is also possible to simply call `opentok.dial(sessionID, token, sipUri)` to start the SIP call.<br/><br/> 
+
+
+The `streamCreated` listener, as shown here in [index.ejs](./views/index.ejs), inspects this token metadata to determine if the new participant is joining via a SIP call:
+
+```javascript
+var session = OT.initSession(sessionId);
+session.on("streamCreated", function (event) {
+  var tokenData = event.stream.connection.data;
+    if (tokenData && tokenData.includes("sip=true")) {
+      var element = "sipPublisherContainer";
+    } else {
+      var element = "webrtcPublisherContainer";
+     }
+
+  . . .
+
+}
+```
+
+
+Once the session has been established and the SIP token (`sipToken`) has been generated, the REST POST request passes that information to the SIP call endpoint:
+
+
+```javascript
+router.post('/sip/start', function(req, res, next) {
+  console.log('received sip start call');
+  var sessionId = req.body.sessionId;
+  var apiKey = req.body.apiKey;
+
+  var options = {
+    uri: 'http://api.opentok.com/v2/project/' + apiKey + '/dial',
+    method: 'POST',
+    json: {
+      sessionId: sessionId,
+      token: sipToken,
+      sip: {
+        uri: config.sipUri
+      },
+      auth: {
+        username: config.sipUsername,
+        password: config.sipPassword
+      },
+      secure: true
+    },
+    headers: {
+      'X-TB-PARTNER-AUTH': apiKey + ':' + config.apiSecret
+    }
+  };
+
+  // Pass in headers if specified
+  if (config.sipHeaders && Object.keys(config.sipHeaders).length !== 0) {
+    options.sip.headers = config.sipHeaders;
+  }
+
+  request(options, function (error, response, body) {
+    if (!error && response.statusCode === 200) {
+      console.log('Successfully started SIP call!');
+      res.send(body);
+    } else {
+      console.log('Failure to start SIP call!');
+      console.log(response);
+      res.status(response.statusCode).send(body);
+    }
+  });
+});
+```
+
```

Comment:
in general, i think this README is really too verbose for anyone to carefully read it all the way through. the more we expand on in the docs, the less likely anyone is to find the content that is actually critical because they tend to glossover it all. i recommend reducing this significantly.

 the structure for the other samples is just 1) Running the App and 2) Walkthrough. i find that getting our audience to read even this much is challenging, but we've struck a balance that seems to be satisfactory.